front/src/modules/timesStops/TimeCell.tsx

```diff
@@ -0,0 +1,230 @@
+import { useState } from 'react';
+
+const TIME_MASK = [
+  { char: '-', isEditable: true, value: 'twentyTens' },
+  { char: '-', isEditable: true, value: 'twentyUnits' },
+  { char: ':', isEditable: false },
+  { char: '-', isEditable: true, value: 'sixtyTens' },
+  { char: '-', isEditable: true, value: 'sixtyUnits' },
+  { char: ':', isEditable: false },
+  { char: '-', isEditable: true, value: 'sixtyTens' },
+  { char: '-', isEditable: true, value: 'sixtyUnits' },
+];
+
+// Converts a raw digit string into a masked time string
+// e.g., "123456" -> "12:34:56"
+const applyMask = (rawDigits: string): string => {
+  let result = '';
+  let digitIndex = 0;
+
+  for (const maskInfo of TIME_MASK) {
+    if (!maskInfo.isEditable) {
+      result += maskInfo.char;
+      continue;
+    }
+
+    if (digitIndex < rawDigits.length) {
+      switch (maskInfo.value) {
+        case 'twentyTens': {
+          const tensDigit = parseInt(rawDigits[digitIndex], 10);
+          if (tensDigit > 2) {
+            result += '2';
+            digitIndex++;
+            continue;
+          }
+          break;
+        }
+        case 'twentyUnits': {
+          const tensDigit = parseInt(result[0], 10);
+          const unitDigit = parseInt(rawDigits[digitIndex], 10);
+          if (tensDigit === 2 && unitDigit > 3) {
+            result += '3';
+            digitIndex++;
+            continue;
+          }
+          break;
+        }
+        case 'sixtyTens': {
+          const tensDigit = parseInt(rawDigits[digitIndex], 10);
+          if (tensDigit > 5) {
+            result += '5';
+            digitIndex++;
+            continue;
+          }
+          break;
+        }
+        default:
+          break;
+      }
+
+      result += rawDigits[digitIndex];
+      digitIndex++;
+    } else {
+      result += maskInfo.char;
+      digitIndex++;
+    }
+  }
+
+  return result;
+};
+
+// Utility to delay execution until after render
+function delayAfterRender(callback: () => void) {
+  queueMicrotask(callback);
+}
+
+// Get the next editable position in the input considering the mask
+const getNextEditablePosition = (pos: number): number => {
+  for (let i = pos; i < TIME_MASK.length; i++) {
+    if (TIME_MASK[i].isEditable) return i;
+  }
+  return TIME_MASK.length;
+};
+
+export default function TimeCell(
+  ctx: {
+    getValue: () => Date | null;
+  },
+  // Let the parent component handle the onChange if provided
+  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void,
```

Comment:
You can make it required and drop `setDisplayValue`
```suggestion
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void,
```